src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -2512,16 +2517,45 @@ private DataObject Copy(NormalizedSnapshotSpanCollection spans)
             {
                 var text = GetText(spans);
                 var blocks = GetTextBlocks(spans);
-                var rtf = _rtfBuilderService.GenerateRtf(spans, TextView);
+                string rtf = null;
+                try
+                {
+                    rtf = GenerateRtf(spans);
+                }
+                catch (OperationCanceledException)
+                {
+                    // Ignore cancellation when doing a copy. The user may not even want RTF text so preventing the normal text from being copied would be overkill.
+                }
                 var data = new DataObject();
                 data.SetData(DataFormats.StringFormat, text);
                 data.SetData(DataFormats.Text, text);
                 data.SetData(DataFormats.UnicodeText, text);
-                data.SetData(DataFormats.Rtf, rtf);
+                if (rtf != null)
+                {
+                    data.SetData(DataFormats.Rtf, rtf);
+                }
                 data.SetData(ClipboardFormat, blocks);
                 return data;
             }
 
+            private string GenerateRtf(NormalizedSnapshotSpanCollection spans)
+            {
+                // This behavior is consistent with VS editor. 
+                // Don't generate RTF for large spans (since it is expensive and probably not wanted).
+                int length = spans.Sum((span) => span.Length);
+                if (length < 1000000)
+                {                                           
+                    using (var dialog = WaitHelper.Wait(_waitIndicator, InteractiveWindowResources.WaitTitle, InteractiveWindowResources.WaitMessage))
```

Comment:
You probably don't actually want the title (the title & message actually display in the message part; the title is always "Microsoft Visual Studio"). Let's make this actually show up before you merge. I think it takes about 3 seconds before the wait indicator shows. Try adding a wait (for testing) for >3 seconds before calling into ```GenerateRtf``` and it should show up.